drivers/timer/Kconfig

```diff
@@ -99,6 +99,7 @@ config CORTEX_M_SYSTICK
 	default $(dt_compat_enabled,$(DT_COMPAT_ARM_V6M_SYSTICK)) || \
 		$(dt_compat_enabled,$(DT_COMPAT_ARM_V7M_SYSTICK)) || \
 		$(dt_compat_enabled,$(DT_COMPAT_ARM_V8M_SYSTICK))
+	depends on !STM32_LPTIM_TIMER
```

Comment:
this should not be done here and instead should be done on the SoC level.

Reply:
Agreed @nashif .
Problem is, as stated above, doing at SoC level doesn't work due to the dts based selection.

Alternative solution is to change dts description each time we want to unable CONFIG_PM, but this doesn't feel right neither.